.github/workflows/gating.yaml

```diff
@@ -104,6 +104,15 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: PyPI test server start
```

Comment:
Shouldn't we just start it right before integration tests?

Reply:
Does it really matter when we start it?

Reply:
To support my argument a bit, the server usually takes a couple of seconds to spin up, so if a container fails to execute (it's CI after all) I want to fail early and not wait until the cachi2 image is built, so I prefer spinning up the servers as soon as possible in the build & integration job.

Reply:
Running process that takes resources when you don't need it matters to me, but I assume it's small.

Reply:
Sure, but running a container image build takes them as well and the image is useless if a test server which is required isn't running ;) , so I'd argue the lesser evil wins -> marking as resolved.